app/rpc.ts

```diff
@@ -367,10 +367,15 @@ export default class RPC {
 
     this.fnSetAddressesWithBalance(addresses);
 
+    // addresses out of the balance.
+    const a = await RPCModule.execute('addresses', '');
+    const aJSON = JSON.parse(a);
+
     // Also set all addresses
+    const allOAddresses =       aJSON.o_addresses.map((o: any) => o);
     const allZAddresses = balanceJSON.z_addresses.map((o: any) => o.address);
     const allTAddresses = balanceJSON.t_addresses.map((o: any) => o.address);
-    const allAddresses = allZAddresses.concat(allTAddresses);
+    const allAddresses = allZAddresses.concat(allTAddresses.concat(allOAddresses));
```

Comment:
This looks surprising...
I would have expected `allAddresses` to successively shadow itself with more concatenations each time.

Reply:
Those three are arrays with 1 or N elements, with the nested concat I merge the three arrays. Another way to do this could be with the spread operator (...), something like this:
`const allAddresses = [...allZAddresses, ...allTAddresses, ...allOAddresses];`

However in some future I going to change this, when I can access to Orchard adds through balance, i don't need to read `addresses` here. I don't if I'm asking your question.

Reply:
I think that the spread operator is provides for a readable expression.   Let's use it here!

Reply:
done!